### PR TITLE
Pulsar standalone does not read zk port form conf/standalone.conf

### LINKED ISSUE #3788: Pulsar standalone does not read zk port form conf/standalone.conf
**Describe the bug**
When we run the order bin/pulsar standalone, it does not read the zookeeper port form the file in conf/standalone.conf.

**To Reproduce**
1. Modify the file in conf/standalone.conf. Add zookeeperServers like this: 
zookeeperServers=127.0.0.1:2182
2. run the order: sh bin/pulsar standalone 
3. See error.
09:59:20.184 [main] INFO  org.apache.zookeeper.server.ZooKeeperServer - Server environment:user.dir=/usr/local/pulsar/apache-pulsar-2.4.0-SNAPSHOT
09:59:20.204 [main] INFO  org.apache.zookeeper.server.ZooKeeperServer - Created server with tickTime 3000 minSessionTimeout 6000 maxSessionTimeout 60000 datadir /usr/local/pulsar/apache-pulsar-2.4.0-SNAPSHOT/data/standalone/zookeeper/version-2 snapdir /usr/local/pulsar/apache-pulsar-2.4.0-SNAPSHOT/data/standalone/zookeeper/version-2
09:59:20.225 [main] INFO  org.apache.zookeeper.server.NIOServerCnxnFactory - binding to port 0.0.0.0/0.0.0.0:2181

**Expected behavior**
The priority is args > standalone.conf > default (args is the order “--zookeeper-port”). It means:
1. If the user uses “--zookeeper-port” to run pulsar in standalone (order: sh bin/pulsar standalone --zookeeper-port 2183), the zk port should be 2183. It does not care the zookeeperServers setting in conf/stanalone.
2. If the user run pulsar in standalone straightly (order: sh bin/pulsar standalone), after the user sets zookeeperServers in conf/stanalone ( like this zookeeperServers=127.0.0.1:2182 ) . The zk port should be 2182
3. If the user run pulsar in standalone straightly (order: sh bin/pulsar standalone), and does not set zookeeperServers in conf/stanalone. The zk port should be default port 2181.

**Desktop**
 - OS: Linux

### PR DESCRIPTION
**Pull request by lsy180829**
### Motivation
Fixes #3788

### Modifications
Add codes make it read zk port from standalone.conf. And it follow the Priority: args > conf > default

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java [hunk 1]
```diff
@@ -71,8 +72,20 @@ public PulsarStandaloneStarter(String[] args) throws Exception {
         }
 
         // Set ZK server's host to localhost
-        config.setZookeeperServers(zkServers + ":" + this.getZkPort());
-        config.setConfigurationStoreServers(zkServers + ":" + this.getZkPort());
+        // Priority: args > conf > default
+        if ( argsContains(args,"--zookeeper-port") ) {
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java]:
```suggestion
        if (argsContains(args,"--zookeeper-port")) {
```

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java [hunk 2]
```diff
@@ -71,8 +72,20 @@ public PulsarStandaloneStarter(String[] args) throws Exception {
...
+            config.setZookeeperServers(zkServers + ":" + this.getZkPort());
+        } else {
+            if (config.getZookeeperServers() != null) {
+                this.setZkPort(Integer.parseInt(config.getZookeeperServers().split(":")[1]));
+            }
+            config.setZookeeperServers(zkServers + ":" + this.getZkPort());
+        }
+
+        if ( config.getConfigurationStoreServers() == null ) {
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java]:
```suggestion
        if (config.getConfigurationStoreServers() == null) {
```

**Comment by jiazhai** [comment]
run integration tests
run java8 tests

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java (hunk 2 above)
Comment (lsy180829) [reply pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java]: Changed, thanks!

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java (hunk 1 above)
Comment (lsy180829) [reply pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java]: Done, thanks!

**Comment by lsy180829** [comment]
@jiazhai Changed, PTAL, thanks!

**Pull request merged**